pyterrier/index.py

```diff
@@ -253,6 +253,27 @@ class IndexingType(enum.Enum):
     SINGLEPASS = 2 #: A single-pass indexing regime, which builds an inverted index directly. No direct index structure is created. Typically is faster than classical indexing.
     MEMORY = 3 #: An in-memory index. No direct index is created.
 
+class TerrierStemmer(Enum):
+    none = 'none'
+    porter = 'porter'
+    weakporter = 'weak'
+
+    @staticmethod
+    def to_class(this):
+        if this is None or this == TerrierStemmer.none:
```

Comment:
How about adding

```python
try:
    this = TerrierStemmer(this)
except ValueError:
    pass
```

which will allow users to either pass in e.g., either `TerrierStemmer.porter` or `"porter"`

Reply:
not sure I follow. it was intended to be a private method anyway. perhaps it should start with `_`

Reply:
I mean so that the user of the indexer doesn't need to fully qualify the enum. E.g.,

```python
indexer = IterDictIndexer(..., stemmer="porter")
```